### PR TITLE
refactor: post masonry

### PR DESCRIPTION
**Pull request by rohitp-bsf**
### Description
- refactor: post masonry

### Screenshots
- NA

### Types of changes
- Converted Class-based components to functional-based.
- Refactored the edit JS file into render and settings.

### How has this been tested?
- NA

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/post/post-masonry/block.js [hunk 1]
```diff
@@ -3,34 +3,34 @@
...
-import ".././editor.scss"
+import '.././style.scss';
+import '.././editor.scss';
 
 // Components
 import { __ } from '@wordpress/i18n';
 
 // Register block controls
-const { registerBlockType } = wp.blocks
+const { registerBlockType } = wp.blocks;
```
Comment (sandeshjangam) [inline src/blocks/post/post-masonry/block.js:17]: import { registerBlockType } from '@wordpress/blocks';

File: src/blocks/post/post-masonry/edit.js [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
-import styling from ".././styling"
-import {
-	InnerBlockLayoutContextProvider,
-	DEFAULT_POST_LIST_LAYOUT,
-	getPostLayoutConfig,
-	getBlockMap
-} from '.././function';
+const TypographyControl = lazy( () =>
+	import(
+		/* webpackChunkName: "chunks/post-masonry/settings" */ '../../../components/typography'
```
Comment (sandeshjangam) [inline src/blocks/post/post-masonry/edit.js:13]: Use alias

File: src/blocks/post/post-masonry/edit.js [hunk 3]
```diff
@@ -1,29 +1,38 @@
...
 
-const { compose } = wp.compose
-const { Component, Fragment  } = wp.element
-import { __ } from '@wordpress/i18n';
-const MAX_POSTS_COLUMNS = 8
-const { createBlock } = wp.blocks
-const {
+const ColumnResponsive = lazy( () =>
+	import(
+		/* webpackChunkName: "chunks/post-masonry/settings" */ '../../../components/typography/column-responsive'
```
Comment (sandeshjangam) [inline src/blocks/post/post-masonry/edit.js:19]: Use alias

File: src/blocks/post/post-masonry/settings.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
+import WebfontLoader from '../../../components/typography/fontloader';
```
Comment (sandeshjangam) [inline src/blocks/post/post-masonry/settings.js:1]: use alais

**Review by sandeshjangam — CHANGES_REQUESTED** [review]
Need changes

File: src/blocks/post/post-masonry/block.js (hunk 1 above)
Comment (rohitp-bsf) [reply src/blocks/post/post-masonry/block.js]: Done!

File: src/blocks/post/post-masonry/edit.js (hunk 2 above)
Comment (rohitp-bsf) [reply src/blocks/post/post-masonry/edit.js]: Done!

File: src/blocks/post/post-masonry/edit.js (hunk 3 above)
Comment (rohitp-bsf) [reply src/blocks/post/post-masonry/edit.js]: Done!

File: src/blocks/post/post-masonry/settings.js (hunk 4 above)
Comment (rohitp-bsf) [reply src/blocks/post/post-masonry/settings.js]: Done!

File: src/blocks/post/post-masonry/edit.js [hunk 5]
```diff
@@ -993,438 +1577,907 @@ class UAGBPostMasonry extends Component {
...
-				{ loadMetaGoogleFonts }
-				{ loadExcerptGoogleFonts }
-				{ loadCtaGoogleFonts }
-			</Fragment>
-		)
+				<Placeholder
+					icon="admin-post"
+					label={
+						uagb_blocks_info.blocks[ 'uagb/post-masonry' ].title
+					}
```
Comment (RajkiranBagal) [inline src/blocks/post/post-masonry/edit.js:2359]: Remove the old return & settings code above.

File: src/blocks/post/post-masonry/edit.js [hunk 6]
```diff
@@ -993,438 +1577,907 @@ class UAGBPostMasonry extends Component {
...
-	let rest_base = ""
+		const deviceType = __experimentalGetPreviewDeviceType
+			? __experimentalGetPreviewDeviceType()
+			: null;
 
-	if ( "undefined" != typeof currentTax ) {
+		const allTaxonomy = uagb_blocks_info.all_taxonomy;
+		const currentTax = allTaxonomy[ postType ];
+		let categoriesList = [];
+		let rest_base = '';
```
Comment (RajkiranBagal) [inline src/blocks/post/post-masonry/edit.js:2412]: camelCase variables

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

**Review by sandeshjangam — APPROVED** [review]

File: src/blocks/post/post-masonry/edit.js (hunk 6 above)
Comment (rohitp-bsf) [reply src/blocks/post/post-masonry/edit.js]: Done!

File: src/blocks/post/post-masonry/edit.js (hunk 5 above)
Comment (rohitp-bsf) [reply src/blocks/post/post-masonry/edit.js:2359]: Done!

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**